core/business.go

```diff
@@ -186,10 +203,39 @@ func (d *Dispatcher) Run() {
 		worker.Start()
 	}
 
-	go d.dispatch()
+	RequestQueue = InitHeap()
+	go d.store()
```

Comment:
Why here did you change d.dispatch to d.store. The intent of d.dispatch was to provide transparent pipe to dispatch jobs, i.e. to handle the load. The job can do anything, e.g. perform transfer or store request to the heap. I would suggest to reverse back to d.dispatch and within Worker Start function instead of calling job.TransferRequest.Run() you may store request to the heap. If you'll have generic interface which will do store/retrieve functionality you'll keep this code mostly un-touched.